### PR TITLE
Refactoring and cleanup

### PR DESCRIPTION
**Pull request by miiu96**
- Added a factory for the `WebSocket` host.
- Removed the driver implementation.
- Removed the unused stubs and interfaces.

### CODE REVIEW

File: websocket/factory/factoryHost.go [hunk 1]
```diff
@@ -4,22 +4,20 @@ import (
...
 	outportData "github.com/multiversx/mx-chain-communication-go/websocket/data"
-	"github.com/multiversx/mx-chain-communication-go/websocket/driver"
 	"github.com/multiversx/mx-chain-communication-go/websocket/server"
 	"github.com/multiversx/mx-chain-core-go/core"
 	"github.com/multiversx/mx-chain-core-go/marshal"
 )
 
-// ArgsWebSocketDriverFactory holds the arguments needed for creating a webSocketsDriverFactory
-type ArgsWebSocketDriverFactory struct {
+type ArgsWebSocketHost struct {
```
Comment (iulianpascalau) [inline websocket/factory/factoryHost.go:12]: missing comment

File: websocket/factory/factoryHost.go [hunk 2]
```diff
@@ -4,22 +4,20 @@ import (
...
 	WebSocketConfig outportData.WebSocketConfig
 	Marshaller      marshal.Marshalizer
 	Log             core.Logger
 }
 
-// NewWebSocketDriver will handle the creation of all the components needed to create an outport driver that sends data over WebSocket
-func NewWebSocketDriver(args ArgsWebSocketDriverFactory) (websocket.Driver, error) {
-	var host websocket.HostWebSocket
+// CreateWebSocketHost will create and start a new instance of websocket.FullDuplexHost
```
Comment (iulianpascalau) [inline websocket/factory/factoryHost.go:18]: a factory.FullDuplexHost *

Comment (miiu96) [reply websocket/factory/factoryHost.go:18]: done.

File: websocket/factory/factoryHost.go [hunk 3]
```diff
@@ -33,17 +32,10 @@ func NewWebSocketDriver(args ArgsWebSocketDriverFactory) (websocket.Driver, erro
...
 	host.Start()
 
-	return driver.NewWebsocketDriver(
-		driver.ArgsWebSocketDriver{
-			Marshaller:      args.Marshaller,
-			WebsocketSender: host,
-			Log:             args.Log,
-		},
-	)
+	return host, nil
```
[the remark this answers is not in the document]
Comment (miiu96) [reply websocket/factory/factoryHost.go:33]: refactored this part.

**Pull request merged**